src/software/ai/navigator/navigator.cpp

```diff
@@ -44,56 +49,59 @@ void Navigator::visit(const KickIntent &kick_intent)
+    if ((path_points).size() > 2)
     {
-        planned_paths.emplace_back(*path_points);
-        if ((*path_points).size() > 2)
-        {
-            current_destination = (*path_points)[1];
-            double segment_final_vel =
-                getCloseToEnemyObstacleFactor((*path_points)[1]) *
-                calculateTransitionSpeedBetweenSegments(
-                    (*path_points)[0], (*path_points)[1], (*path_points)[2],
-                    ROBOT_MAX_SPEED_METERS_PER_SECOND *
-                        Util::DynamicParameters::Navigator::transition_speed_factor
-                            .value());
+        current_destination = (path_points)[1];
+        double segment_final_vel =
+            getCloseToEnemyObstacleFactor((path_points)[1]) *
+            calculateTransitionSpeedBetweenSegments(
+                (path_points)[0], (path_points)[1], (path_points)[2],
+                ROBOT_MAX_SPEED_METERS_PER_SECOND *
+                    Util::DynamicParameters::Navigator::transition_speed_factor.value());
 
-            auto move = std::make_unique<MovePrimitive>(
-                move_intent.getRobotId(), current_destination,
-                move_intent.getFinalAngle(), segment_final_vel,
-                move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
-                move_intent.getAutoKickType());
-            current_primitive = std::move(move);
-            return;
-        }
-        if ((*path_points).size() == 2)
-        {
-            current_destination = (*path_points)[1];
-            auto move           = std::make_unique<MovePrimitive>(
-                move_intent.getRobotId(), current_destination,
-                move_intent.getFinalAngle(),
-                move_intent.getFinalSpeed() *
-                    getCloseToEnemyObstacleFactor((*path_points)[1]),
-                move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
-                move_intent.getAutoKickType());
-            current_primitive = std::move(move);
-            return;
-        }
+        auto move = std::make_unique<MovePrimitive>(
+            move_intent.getRobotId(), current_destination, move_intent.getFinalAngle(),
+            segment_final_vel, move_intent.isDribblerEnabled(),
+            move_intent.isSlowEnabled(), move_intent.getAutoKickType());
+        current_primitive = std::move(move);
+    }
+    else if ((path_points).size() == 2)
+    {
+        current_destination = (path_points)[1];
+        auto move           = std::make_unique<MovePrimitive>(
+            move_intent.getRobotId(), current_destination, move_intent.getFinalAngle(),
+            move_intent.getFinalSpeed() * getCloseToEnemyObstacleFactor((path_points)[1]),
+            move_intent.isDribblerEnabled(), move_intent.isSlowEnabled(),
+            move_intent.getAutoKickType());
+        current_primitive = std::move(move);
+    }
+    else
+    {
+        throw std::runtime_error("Path only contains one point.");
```

Comment:
I'm confused, why does a 1 point path throw an exception? Shouldn't we just move straight to that point?

Reply:
all paths have a start point and an end point, so it a one point path defining a start point or an end point?

Reply:
Logically I would assume that it means both the start and end point (ie. the robot should navigate to that point and stop if it's not already there).

Reply:
Any valid path _must_ contain a start and destination; this is the interface that navigator expects. I added a comment in the path planner interface

Reply:
why? I'm still confused why we can't treat a single point path as both the start and end point?

Reply:
how is treating a single point as both start and dest a good idea? navigator will still have to process that into two points, so it'll still have a special case for the single point vector. It might also hide problems in the path planner, where it somehow misses a point

Reply:
I kind of like having 2 points returned. It's a more consistent interface that reduces the number of checks the navigator makes, as basically no extra cost to the planner. We can sync up in person on Saturday.

Reply:
Mmk, I don't feel super strongly about this one either way, as long as we're clearly doc'ing these requirements in interfaces.